src/Common/src/System/Collections/Generic/LowLevelDictionary.IDictionary.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal partial class LowLevelDictionary<TKey, TValue> : IDictionary
+    {
+        bool IDictionary.IsFixedSize => false;
+        bool IDictionary.IsReadOnly => false;
+
+        object IDictionary.this[object key]
+        {
+            get { return this[(TKey)key]; }
+            set { this[(TKey)key] = (TValue)value; }
+        }
+
+        ICollection IDictionary.Keys
```

Comment:
FYI, if you do end up switching to Dictionary, you may want to take a look at what I did for `SqlConnection.RetreiveStatistics`, which previously returned Hashtable but now returns Dictionary (to break the dependency on the non-generic collections). I had to subclass Dictionary to make it fully compatible with Hashtable.

https://github.com/dotnet/corefx/blob/a59bf6473733827293030457c8a8b147f1e573af/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlStatistics.cs#L225-L333